### PR TITLE
support volume guid

### PR DESCRIPTION
**Pull request by YuezhangMo**
The updates of this pull request.

- support get first empty dentry set
- add option -U to mkfs.exfat
- add option -U/-u to tune.exfat
- fsck.exfat no longer remove GUID entry

### CODE REVIEW

File: lib/libexfat.c [hunk 1]
```diff
@@ -494,6 +494,166 @@ int exfat_set_volume_label(struct exfat *exfat, char *label_input)
...
+	return 0;
+}
+
+int exfat_read_volume_guid(struct exfat *exfat)
+{
+	int err;
+	uint16_t checksum = 0;
+	struct exfat_dentry *dentry;
+	struct exfat_lookup_filter filter = {
+		.in.type = EXFAT_GUID,
```
Comment (hclee) [inline lib/libexfat.c:559]: it is better to initialize in.dentry_count also. otherwise This PR looks good to me.

Comment (YuezhangMo) [reply lib/libexfat.c:559]: Thank you for your comment.
I added initialization of in.dentry_count.

**Comment by namjaejeon** [comment]
Applied, Thanks for your patch!

**Comment by namjaejeon** [comment]
@YuezhangMo I have a question, Is there any reason why you choose -u(and -U) option instead of -g(and -G) option for guid support ?

**Comment by YuezhangMo** [comment]
> @YuezhangMo I have a question, Is there any reason why you choose -u(and -U) option instead of -g(and -G) option for guid support ?

GUID is an implementation of UUID. Other Linux file systems support UUID and use the -U option, such `mkfs.btrfs -U` and `mkfs.ext4 -U`.

**Pull request closed without merging**